### PR TITLE
Refactor Simple KVBC test

### PR DESCRIPTION
**Pull request by salieri11**
This PR enables the Simple KVBC test to accept new optional CLI parameter, -n, that is the name of the communication configuration file. This change will enable the user to run this test on different machines. (see #52)
Moreover, 2 explicit replicas parameters were removed : -f and -c. These parameters are taken from parsing the replica's key file.
Lastly, Simple KVBC test was refactored to use common structs with the Simple test.

### CODE REVIEW

File: bftengine/tests/simpleKVBCTests/TesterReplica/main.cpp [hunk 1]
```diff
@@ -67,88 +97,80 @@ int main(int argc, char **argv) {
...
-
-	if(repId == UINT16_MAX || 
-		 keysFile.empty() ||
-		 fVal == UINT16_MAX ||
-		 cVal == UINT16_MAX
-		)
+	if(rp.replicaId == UINT16_MAX || rp.keysFilePrefix.empty())
 	{
-		fprintf(stderr, "%s -k KEYS_FILE_PREFIX -f F -c C -i ID", argv[0]);
+		fprintf(stderr, "%s -k KEYS_FILE_PREFIX -i ID -f COMM_CONFIG_FILE",argv[0]);
```
Comment (beerriot) [inline bftengine/tests/simpleKVBCTests/TesterReplica/main.cpp]: typo: should be "-n COMM_CONFIG_FILE", not "-f"

File: bftengine/tests/simpleKVBCTests/TesterClient/main.cpp [hunk 2]
```diff
@@ -28,24 +31,39 @@
...
+	char argTempBuffer[PATH_MAX + 10];
+	ClientParams cp;
+	cp.clientId = UINT16_MAX;
+	cp.numOfFaulty = UINT16_MAX;
+	cp.numOfSlow = UINT16_MAX;
+	cp.numOfOperations = UINT16_MAX;
 
 	int o = 0;
-	while ((o = getopt(argc, argv, "i:f:c:p:")) != EOF) {
+	while ((o = getopt(argc, argv, "i:f:c:p:n:")) != EOF) {
```
Comment (beerriot) [inline bftengine/tests/simpleKVBCTests/TesterClient/main.cpp]: Why are f and c still needed for the client process? Shouldn't these come from the config file now?

File: bftengine/tests/simpleKVBCTests/TesterClient/main.cpp [hunk 3]
```diff
@@ -90,51 +108,66 @@ int main(int argc, char **argv) {
 			string numOfOpsStr = argTempBuffer;
 			int tempfVal = std::stoi(numOfOpsStr);
 			if (tempfVal >= 1 && tempfVal < UINT32_MAX)
-				numOfOps = (uint32_t)tempfVal;
+				cp.numOfOperations = (uint32_t)tempfVal;
 			// TODO: check numOfOps
 		}
 		break;
 
+        case 'n':
```
Comment (beerriot) [inline bftengine/tests/simpleKVBCTests/TesterClient/main.cpp]: nit: align with surrounding code please

File: bftengine/tests/simpleKVBCTests/TesterReplica/main.cpp [hunk 4]
```diff
@@ -67,88 +97,80 @@ int main(int argc, char **argv) {
...
-			argTempBuffer[sizeof(argTempBuffer) - 1] = 0;
-			string cStr = argTempBuffer;
-			int tempcVal = std::stoi(cStr);
-			if (tempcVal >= 0 && tempcVal < UINT16_MAX)
-				cVal = (uint16_t)tempcVal;
-			// TODO: check cVal
-		}
-		break;
+		case 'n':
+        {
```
Comment (beerriot) [inline bftengine/tests/simpleKVBCTests/TesterReplica/main.cpp]: nit: align with surrounding code, please

File: bftengine/tests/simpleKVBCTests/TesterClient/main.cpp (hunk 2 above)
Comment (salieri11) [reply bftengine/tests/simpleKVBCTests/TesterClient/main.cpp]: Today f and c are defined from replica keys configuration file and the client doesn't need it.
When we will switch to full configuration file for the concord nodes - these parameters will be cleaned from the CLI.

Comment (salieri11) [reply bftengine/tests/simpleKVBCTests/TesterClient/main.cpp]: Today f and c are defined from replica keys configuration file and the client doesn't need it.
When we will switch to full configuration file for the concord nodes - these parameters will be cleaned from the CLI.

File: bftengine/tests/simpleKVBCTests/TesterReplica/main.cpp (hunk 1 above)
Comment (salieri11) [reply bftengine/tests/simpleKVBCTests/TesterReplica/main.cpp]: oops, thanks!

**Pull request merged**